lib/tasks/one_time_use/recreate_timeout_jobs.rake

```diff
@@ -0,0 +1,135 @@
+  job_exists = scheduled_set.each do |job|
+    break true if job['args'].first >= {
+      "job_class" => "UserSessionTimeoutJob",
+      "arguments" => [user_session.id]
+    }
+  end || false
+
+  UserSessionTimeoutJob.set(wait_until: timeout_datetime).perform_later(user_session.id) unless job_exists
+end
+
+def finish_due_classic_user_session(user_session, timeout_datetime)
+  user_session.update(finished_at: timeout_datetime)
+  user_session.send(:delete_alternative_answers)
+  user_session.reload
+
+  user_session.send(:decrement_audio_usage)
+
+  RecreateUserSessionScheduleService.new(user_session, timeout_datetime).schedule
+  V1::ChartStatistics::CreateForUserSession.call(user_session)
+
+  AfterFinishUserSessionJob.perform_later(user_session.id, user_session.session.intervention)
+
+  user_session.update_user_intervention(session_is_finished: true)
+end
+
+def finish_due_catmh_user_session(user_session, timeout_datetime)
+  user_session.update!(finished_at: timeout_datetime)
+
+  cat_mh_api = Api::CatMh.new
+
+  result = cat_mh_api.get_result(user_session)
+  result_to_answers(user_session, result['body']) if result['status'] == 200
+  cat_mh_api.terminate_intervention(user_session)
+
+  RecreateUserSessionScheduleService.new(user_session, timeout_datetime).schedule
+  GenerateUserSessionReportsJob.perform_later(user_session.id)
+
+  V1::ChartStatistics::CreateForUserSession.call(user_session)
+
+  user_session.update_user_intervention(session_is_finished: true)
+end
+
+def scheduled_set
+  @scheduled_set ||= Sidekiq::ScheduledSet.new
+end
+
+def result_to_answers(user_session, result)
+  available_test_types = user_session.session.cat_mh_test_types
+  result['tests'].each do |test|
+    test_type = available_test_types.find_by(short_name: test['type'].downcase)
+    test_type.cat_mh_test_attributes.each do |variable|
+      Answer::CatMh.create!(
+        user_session_id: user_session.id,
+        body: {
+          'data' => [
+            { 'var' => "#{test_type.short_name}_#{variable.name}", 'value' => test[variable.name] }
+          ]
+        }
+      )
+    end
+  end
+end
+
+class RecreateUserSessionScheduleService < V1::UserSessionScheduleService
+  def initialize(user_session, now)
+    @now = now # define your own now
+    super(user_session)
+  end
+
+  attr_reader :user_session, :now, :all_var_values, :all_var_values_with_session_variables, :health_clinic
+  attr_accessor :next_user_session, :user_intervention
+
+  def after_fill_schedule(next_session)
+    next_user_session.update!(scheduled_at: now)
+    next_session.send_link_to_session(user_session.user, health_clinic)
+  end
+
+  def days_after_fill_schedule(next_session)
+    user_intervention.update!(status: :schedule_pending)
+    next_user_session.update!(scheduled_at: (now + next_session.schedule_payload.days))
+    # SessionScheduleJob.set(wait: next_session.schedule_payload.days).perform_later(next_session.id, user_session.user.id, health_clinic)
```

Comment:
why is this line of code commented out?

Reply:
I thought it might be helpful to leave a reminder that it was there, where the job should have been scheduled (to make it easier to understand what do we want to accomplish here)